### DISCUSSION TITLE
Lerna removing from vs updating lockfile

### DISCUSSION THREAD

**Discussion by wyardley** [renovatebot/renovate#23567] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

As of recently, Renovate is sometimes not giving an artifact error, but is still removing, vs. updating, the package in the `package-lock.json` when updating pinned packages in a repo that uses Lerna.

https://github.com/wyardley/renovate-repro-lerna/
e.g.,
https://github.com/wyardley/renovate-repro-lerna/pull/1

This might be relevant (from the debug logs; longer snippet below):
```
DEBUG: Finding package.json for lerna location "lerna.json"
```
maybe it's trying to use `lerna.json` as `package.json` for the root install? That contains the lerna config, but not the lerna version (which is in the root `package.json`)

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Finding package.json for lerna location "lerna.json" (branch="renovate/major-lerna-monorepo")
DEBUG: npmrc file found in repository (branch="renovate/major-lerna-monorepo")
DEBUG: Spawning lerna with npm to create lock files (branch="renovate/major-lerna-monorepo")
DEBUG: No node constraint found - using latest (branch="renovate/major-lerna-monorepo")
WARN: Could not detect lerna version in package.json, using 'latest' (branch="renovate/major-lerna-monorepo")
DEBUG: Skipping lerna bootstrap (branch="renovate/major-lerna-monorepo")
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by wyardley** [comment]
The [issues dash](https://github.com/wyardley/renovate-repro-lerna/issues/3) is warning that it can’t detect the lerna version and is using latest, even though it’s [specified](https://github.com/wyardley/renovate-repro-lerna/blob/c09c06ad6582452558a818a4233833531f4c6ae6/package.json#L13) in the top level `package.json`. So this may be due to it using 7 which expects you to be using npm workspaces?

<img width="896" alt="image" src="https://github.com/renovatebot/renovate/assets/10714426/f8a70ce0-2295-4c8f-8d69-569457d9a9eb">

**Comment by rarkins** [comment, marked as answer]
@viceice I think this broke with last week's lazy detection change. It looks for lerna version only in packageManager and engines, instead of dependencies and devDependencies: https://github.com/renovatebot/renovate/blame/c69247c9db9b6031ab17ea824bce53ce398094f7/lib/modules/manager/npm/post-update/lerna.ts#L29

Fix in #23575

**Comment by wyardley** [comment]
Thanks!
